### ISSUE TITLE
Scikit-learn wrapper score method fails if a non-generic accuracy measure is specified.

### ISSUE THREAD

**Issue by JimBiardCics** [fchollet/keras#5241]
If a scikit-learn wrapper KerasClassifier object is created using a model that is compiled with any metrics argument list that does not contain the string 'accuracy' (say, you specify a specific accuracy metric, such as 'categorical_accuracy', or a custom accuracy function), the score method will raise an exception complaining that no accuracy metric was supplied. This occurs in lines 252-257 of keras/keras/wrappers/scikit_learn.py. There needs to be a way to specify a specific accuracy function or name. Here's a code snippet that will reproduce the problem:

```
def modelbuild(input_shape = (301, 301, 1), lr = lr):
    model = Sequential

    model.add(Convolution2D(input_shape = input_shape, nb_filters = 32, nb_col = 5, nb_row = 5, activation = 'relu'))
    model.add(Flatten())
    model.add(Dense(4, activation = 'softmax'))

    model.compile(loss = 'categorical_crossentropy', metrics = ['categorical_accuracy'], optimizer = Adam(lr = lr))

    return model

sk_params = {
    'input_shape' : (301, 301, 1),
    'lr' : 0.001
}

classifier = KerasClassifier(modelbuild, **sk_params)

param_grid = {
    'lr' : [0.001, 0.002, 0.003]
}

searchObj = GridSearchCV(classifier, param_grid = param_grid, n_jobs = 1)

searchObj.fit(inputArray, truthArray)
```
